docs/decisions/adr-extend-application-manifest.md

```diff
@@ -1,79 +1,120 @@
 
-<!-- This is an optional element. Feel free to remove. -->
-## More Information
+## Validation
 
-{You might want to provide additional evidence/confidence for the decision outcome here and/or
- document the team agreement on the decision and/or
- define when and how this decision should be realized and if/when it should be re-visited and/or
- how the decision is validated.
- Links to other decisions and resources might appear here as well.}
\ No newline at end of file
+The implementation of both API endpoint and extended application information will be validated
+through additional API and command tests added to the testsuites used by the CI/CD system.
+
+## Specification Details
+
+### API endpoint
+
+The new API endpoint is an extension of the existing `AppPart` endpoint, i.e. of
+`/namespaces/:namespace/applications/:app/part/:part`.
+
+This endpoint is extended with a new allowed part value `manifest`, which, when requested delivers a
+`text/plain` result containing the application manifest serialized into its final form.
+
+The chosen form is the YAML-formatted manifest currently written by the Epinio CLI.
+
+### Extended Application CRD
+
+The extended application information managed by the Web UI revolves around the application origin,
+where the Web UI provides more features to the user than the CLI.
+
+The existing `origin` fields in the CRD are for the handling of paths, containers, and git
+references.
+
+In the case of paths the Web UI enables the user to make use of a variety of archive files, whereas
+the CLI only supports directories. To this end a new field `archive` is added to the `origin`
+object, of type `string`. This field is exptected to contain values like `zip`, `tarball`, `tar+gz`,
+etc. Note however that this list of examples is not considered to be exhaustive and is definitely
+not a list of the only allowed values for the field.y
+
+No additional information is managed for containers.
+
+For git references two additional fields are added to the `git` objects in the `origin`
+object. These fields are `provider` and `branch`, both of type `string`.
+
+The first field, `provider` provides (sic!) information about the provider of the git repository,
+with expected values of `git`, `github`, and `gitlab`. Note however that this list of examples is
+not considered to be exhaustive and is definitely not a list of the only allowed values for the
+field.
+
+The second field, `branch`, records the name of the repository branch the `revision` is part of, in
+support of the Web UIs ability to ....
+
+The modified part of the CRD is
+
+```
+origin:
+  properties:
+    archive:
+      type: string
+    container:
+      type: string
+    git:
+      properties:
+        branch:
+          type: string
+        provider:
+          type: string
+        repository:
+          type: string
+        revision:
+          type: string
+      required:
+      - repository
```

Comment:
from the ui side, not sure of cli, the revision will always be required as well

Reply:
On the cli side the revision is optional. When left out the server's git client uses the HEAD, afaik.